### DISCUSSION TITLE
tools.otenet.gr: nuisance

### DISCUSSION THREAD

**Discussion by DimitrisKalaboukas** [uBlockOrigin/uAssets#16400] [Q&A] [answered]
### Prerequisites

- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I forced an update of my filter lists. (Click the ["Purge all caches"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches) button while holding the 'Shift' key, then click the ["Update now"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now) button.)
- [X] I did not remove any of the [default filter lists](https://user-images.githubusercontent.com/585534/91296476-ab9d8c00-e76a-11ea-9fb9-87d6861bdf73.png), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable external filter lists, or I have verified that the issue still occurs without enabling external filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other [similarly-purposed blockers](https://twitter.com/gorhill/status/1033706103782170625).
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL address of the web page

`https://tools.otenet.gr/?_task=mail`

### Category

nuisance

### Description

Annoying popup wizards for password change and mobile phone number registration that overwhelm the screen every time you log in and can’t be permanently disabled. Because the page is on the deep web (customer account needed), I saved a full HTML copy of each wizard (https://bit.ly/3XmygHr). The following filters:

```adb
tools.otenet.gr##div.ui-dialog:has-text(Καταχώρηση τηλεφωνικού αριθμού)
tools.otenet.gr##div.ui-dialog:has-text(Σημαντική Ενημέρωση)
tools.otenet.gr##div.ui-widget-overlay
```

block the popups and their overlays, but the background page is not fully functional, unless you refresh it or hit “escape”. Could anyone help to optimize the above filters?

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![Screenshot 2023-01-19 at 10-38-48 COSMOTE Webmail Εισερχόμενα](https://user-images.githubusercontent.com/101707608/213397336-23a37fb1-8fda-48c9-b3c7-369f830a8cc2.png)
![Screenshot 2023-01-19 at 10-41-39 COSMOTE Webmail Εισερχόμενα](https://user-images.githubusercontent.com/101707608/213397372-98312f0e-8958-42c1-b793-4dd6ecebba3f.png)

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.46.0
Firefox: 108
filterset (summary): 
  network: 210757
  cosmetic: 242979
  scriptlet: 37740
  html: 995
listset (total-discarded, last-updated): 
  added: 
    https://raw.githubusercontent.com/DandelionSprout/adfilt/master/AnnoyancesList: 43731-607, 1d.20h.59m
    https://raw.githubusercontent.com/hoshsadiq/adblock-nocoin-list/master/nocoin.txt: 1062-0, 20h
    https://secure.fanboy.co.nz/fanboy-problematic-sites.txt: 1955-7, 1d.20h.59m
    https://www.i-dont-care-about-cookies.eu/abp/: 25703-5, 1d.20h.59m
    adguard-generic: 67711-6026, 1d.21h.44m
    adguard-mobile: 8681-187, 1d.21h.44m
    adguard-spyware: 36778-14939, 1d.21h.44m
    adguard-spyware-url: 885-5, 1d.21h.44m
    block-lan: 44-0, 22d.1h.32m
    curben-phishing: 21329-2, 19h.57m
    curben-pup: 187-0, 19h.56m
    adguard-annoyance: 62545-5566, 1d.18h.15m
    adguard-social: 19608-1119, 1d.21h.44m
    fanboy-thirdparty_social: 69-0, 5d.22h.23m
    fanboy-annoyance: 85025-69101, 5d.4m
    fanboy-cookiemonster: 44825-9428, 5d.3m
    fanboy-social: 23494-31, 20h.1m
    ublock-annoyances: 4729-6, 1d.21h.44m
    dpollock-0: 11397-474, 2d.23h.45m
  default: 
    user-filters: 0-0, never
    ublock-filters: 32590-132, 1d.21h.44m
    ublock-badware: 4309-0, 1d.21h.44m
    ublock-privacy: 304-0, 2d.2m
    ublock-abuse: 77-4, 2d.1m
    ublock-unbreak: 1936-0, 1d.21h.44m
    ublock-quick-fixes: 494-1, 19h.58m
    easylist: 62220-1011, 2d
    easyprivacy: 30941-2152, 1d.23h.59m
    urlhaus-1: 6331-0, 19h.55m
    plowe-0: 3668-4, 9d.23h.55m
    GRC-0: 991-2, 19h.59m
filterset (user): [empty]
modifiedUserSettings: [none]
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 16925 ms
  maxAssetCacheWait: 2506 ms
popupPanel: 
  blocked: 1
  blockedDetails: 
    googletagmanager.com: 1
```
</details>

**Comment by DimitrisKalaboukas** [comment]
@krystian3w Could you help?

**Reply by krystian3w** [reply]
My guess is that this may require writing a scriptlet to interrupt the pop-up from running.

---

Before analyzing JavaScript, you can check if removing both elements does something:


```adb
tools.otenet.gr##.ui-widget.ui-dialog:remove()
tools.otenet.gr##.ui-widget-overlay:remove()
```

Then you can make it more specific with `has` or `has-text`.

Perhaps the site supports HTML filtering well and you can on Firefox cut out the pop-up before the browser analyzes the site's tags.

**Comment by u-RraaLL** [comment]
Do you mean you can't scroll?

https://www.reddit.com/r/uBlockOrigin/wiki/solutions/#wiki_locked_scroll_after_blocking_various_overlays

**Reply by krystian3w** [reply]
On my side it's possible to scroll - but saving to local files doesn't seem to have saved all the CSS correctly.

**Reply by DimitrisKalaboukas** [reply]
Not only scrolling but neither clicking on a lot of areas is possible. It looks like a hidden element is causing the issue...

Is there any secret way to send you a test account to log in yourself?

**Reply by krystian3w** [reply]
Not on github - that one doesn't have DM/PM.

Odd that the element picker does not detect the hidden element - could it be a pseudo-element.

To reset the pseudo-element you can use:

```adb
tools.otenet.gr##::before
tools.otenet.gr##::after
```

if they help, then you can review how to narrow the hiding to not accidentally spoil the look of the page (as/if it relies on ugly clearfix CSS techniques: https://css-tricks.com/snippets/css/clear-fix/).

**Comment by DimitrisKalaboukas** [comment]
Unfortunately, these rules don’t work either.

Could you give me a temp email address to send you the test account credentials?

**Comment by DimitrisKalaboukas** [comment]
@krystian3w Is it possible to click automatically on X button of the popups with corresponding rules?

**Reply by stephenhawk8054** [reply]
uBO does not support autoclick, because it's a security-risk.

**Reply by krystian3w** [reply]
Not for now: https://github.com/uBlockOrigin/uBlock-issues/issues/2229 (Point-Of-Click?)

You would have to develop a scriptlet from scratch or fork yourself: https://github.com/NanoAdblocker/NanoFilters/blob/5e63d1a37cdb3e277f8db30e87df56e816c454c9/NanoFilters/NanoResources.txt#L283-L328 (you need to disarm the check to see if the scriptlet has the right to run and update it, if necessary, to the 2023 JS standards)

Very optional can be used PolishCookieConsent with 1 filter stylized to uBo syntax: https://github.com/FiltersHeroes/PolishCookieConsent#how-to-install-it

*(can sometimes fail even though the item is visible or you have set about 300 checks behind the close button - not my fairy tale, and there is no one to fix such problems (excluding frames, there the add-on does not start by default, so it is useless on German sites or tumblr to close cookies/GDPR/CCPA))*

https://polishannoyancefilters.netlify.app/en/PolishCookieConsent/syntax/

```adb
! click element, if the document has finished loading and the document has been
! parsed but sub-resources such as scripts, images, stylesheets and frames are still loading:
tools.otenet.gr##+js(clickInteractive, element/XPath, cookie/count, text)
!
! click element, when it will be available:
tools.otenet.gr##+js(clickTimeout, element/XPath, cookie/count, text)
!
!  click element, when the website will be fully loaded:
tools.otenet.gr##+js(clickComplete, element/XPath, cookie/count, text)
!
```


You can also use one of the thousands of click examples and save as a userscript to, for example, TamperMonkey/ViolentMonkey.

**Comment by DimitrisKalaboukas** [comment]
@krystian3w Unfortunately, I don’t have the knowledge to do so, but I found something else that may help. If you visit `https://tools.otenet.gr/` and click on “Ανάκτηση Συνθηματικού” (at the bottom right), it will open a popup wizard which is similar to the internal ones. Can you block this element retaining the background page scrollable and fully functional?

**Reply by krystian3w** [reply]
Scroll works without any manimpulation of code here ¯\\_(ツ)\_/¯.

But pop-up block on page events like `click`, `mousedown`, `keypress` by script: 
`https://tools.otenet.gr/plugins/jqueryui/js/jquery-ui-1.9.1.custom.min.js` 

(until I myself have to rethink whether I need to undo some blocked jQuery pop-ups (`.ui-dialog`) due to the fact that the filter is harmful without a click neutralizer)

**Reply by krystian3w** [reply]
In PCC you can try:


```adb
tools.otenet.gr##+js(clickInteractive, .ui-dialog-titlebar-close, 300)
tools.otenet.gr##+js(clickTimeout, .ui-dialog-titlebar-close, 300)
tools.otenet.gr##+js(clickComplete, .ui-dialog-titlebar-close, 300)
```

[PCC-my-filters_2023-03-23.txt](https://github.com/uBlockOrigin/uAssets/files/11050601/PCC-my-filters_2023-03-23.txt)

But this can close all ui-dialogs in loop as long you didn't implement Xpath to found text "`Καταχώρηση τηλεφωνικού αριθμού`, `Σημαντική Ενημέρωση`" if page don't "bake" cookies about closed type of pop-up (also cookie name can't be number due PCC parse number as number only).

**Comment by DimitrisKalaboukas** [comment]
@krystian3w Thank you very much for your assistance!

**Reply by krystian3w** [reply]
To regain the clickability of the rest of the interface, sometimes removing an element from the DOM works - before the JS code from the jQuery Dialogs plugin can be executed.

So someone should analyze whether the pop-up HTML code is rigidly embedded in HTML in the template of the first visit after logging in the best without `ui-dialog` classes to asap removing.

If it is injected by the library then `:remove()` may not make it in time and then you should consider checking what function loads pop-up to become independent of a custom cross-click custom scriptlet or safety remove listeners "click", "mousedown", "keypress".

Such removal:
```adb
ekliniki.pl##^#popup-message
! or:
ekliniki.pl###popup-message:remove()
```
works on the Polish site (The jQuery plugin code gets fall and clicks are not blocked) - removes the pop-up base when it has not yet been wrapped with elements from jQuery Dialogs (close button, title bar, standard fancy interface).

**Comment by DimitrisKalaboukas** [comment]
@krystian3w In URLs like:

```
||tools.otenet.gr/?_task=mail&_action=getunread&_remote=1&_unlock=0&_=1694698069145
```

how can I automatically replace the string “_unlock=0” with “_unlock=1”?

**Reply by Yuki2718** [reply]
Replacing URL (suggested as [queryjump](https://github.com/uBlockOrigin/uBlock-issues/issues/760#issuecomment-720080024)) is currently not possible and this certainly is a security risk.

**Reply by MasterKia** [reply]
@DimitrisKalaboukas Add `user-` to `trustedListPrefix` in "Advanced settings" then use:

```
||tools.otenet.gr/?_task=mail&_action=getunread&_remote=1&_unlock=0$urltransform=/_unlock=0/_unlock=1/
```

**Comment by DimitrisKalaboukas** [comment]
@MasterKia This method doesn’t solve the issue.

**Comment by DimitrisKalaboukas** [comment]
@krystian3w If you open:

https://tools.otenet.gr/

and click on “Ανάκτηση Συνθηματικού” at the bottom-right, could you block the popup wizard with a specific ##+js rule?

**Reply by krystian3w** [reply]
No if we still can't detect extra match in `trusted-click-element` due works/worked at my side generic filter with:

```adb
tools.otenet.gr##+js(trusted-click-ekement, .ui-icon-closetick, , 1000)
```

To avoid closing pop-ups for not logged users (or pop-up without unique attributes compatible with CSS 2.1 or CSS4).

I haven't had any success corectly closes the jQuery UI Dialog since I last visited the thread.

**Comment by DimitrisKalaboukas** [comment]
@krystian3w The internal popup wizards can also be partially blocked by the filters:

tools.otenet.gr###oldPasswordPromptMsg:remove()
tools.otenet.gr###pr_popupwizard:remove()

but again some remnants are left behind making the background page dysfunctional. Could they be transformed in a more effective way, e.g. by blocking the popups in a js level?

**Reply by krystian3w** [reply]
I guess not for a long time, because the remove element(s) method would have to allow seeing the page code whether the pop-up without ID maybe is in the page code to remove elements more early (if fails `$doc,replace=/old/new/` or HTML Filtering) with avoid break user select actions.

Perhaps there is no need to think about how to block the popup and implement some filter that fixes clicking, dropping and selecting (by `set`/`aopw`/`aopr`/`aost`/`aeld`) as long works page fine.

**Comment by DimitrisKalaboukas** [comment]
@krystian3w From my side, the rule:

tools.otenet.gr##+js(trusted-click-element, .ui-icon-closetick, , 1000)

doesn’t work even in the external wizard (I have added “user-” in trustedListPrefixes and it is recognized as correct in “My filters”). Also, how can this type of rules be further specified in order not to close useful popups?

**Reply by krystian3w** [reply]
As I understand it, at the moment you would need to detect after what time the good pop-ups pop up and change this 1000 ms to a more correct time value.

The ahead of time parameter probably still does not read cookies or local/session storage, so it is unlikely to be used.

You'd have to play with something like this:
```adb
tools.otenet.gr##+js(trusted-click-element, .ui-dialog.ui-widget:has(#oldPasswordPromptMsg) .ui-icon-closetick, , 3000)
tools.otenet.gr##+js(trusted-click-element, .ui-dialog.ui-widget:has(#pr_popupwizard) .ui-icon-closetick, , 3000)
```

I don't keep track of how low you see both IDs, so they could be next to the ui-dalog class and plain CSS 2.1 would work:

```adb
tools.otenet.gr##+js(trusted-click-element, .ui-dialog.ui-widget#oldPasswordPromptMsg .ui-icon-closetick, , 3000)
tools.otenet.gr##+js(trusted-click-element, .ui-dialog.ui-widget#pr_popupwizard .ui-icon-closetick, , 3000)
```

Also X can by only named as `.ui-dialog-titlebar-close`.

**Comment by DimitrisKalaboukas** [comment]
@D4niloMR If you open:

https://tools.otenet.gr/

and click on “Ανάκτηση Συνθηματικού” at the bottom-right, could you block the popup wizard with a specific ##+js rule?

**Reply by D4niloMR** [reply]
![image](https://github.com/uBlockOrigin/uAssets/assets/70459964/e92929c9-0cfa-4b8b-adc8-47029da19259)

Where is this?

**Reply by DimitrisKalaboukas** [reply]
@D4niloMR At the bottom-right of login form.

![Screenshot 2024-01-18 at 16-48-15 COSMOTE Webmail Καλώς ήρθατε στο COSMOTE Webmail](https://github.com/uBlockOrigin/uAssets/assets/101707608/e51b36d9-2d0a-4993-a2ae-617692c39517)

You can also copy-paste the string in page search box.

**Reply by D4niloMR** [reply]
So, you want to block the password reset popup? but why using a ##+js filter?

**Reply by DimitrisKalaboukas** [reply]
@D4niloMR Not exactly. I want to block two similar internal popup wizards, but the hiding rules:

tools.otenet.gr##.ui-dialog:has(#oldPasswordPromptMsg)
tools.otenet.gr##.ui-dialog:has(#pr_popupwizard)
tools.otenet.gr##.ui-widget-overlay

partially break the background page, which cannot be fixed with position and overflow resetting rules, so I need to prevent them from even popping up.

**Reply by D4niloMR** [reply]
I can't see any of this. But anyway, some members already tried, it's unlikely that I will come up with a solution.

**Reply by D4niloMR** [reply]
Put these filters in "My filters" and see if it has relevant info:

```
tools.otenet.gr##+js(nostif)
tools.otenet.gr##+js(nosiif)
tools.otenet.gr##+js(aeld, _, , log, 3)
```

You might need to set `filterAuthorMode` to "true" in the advanced settings.

**Reply by DimitrisKalaboukas** [reply]
@D4niloMR These filters don’t help. You should be logged in with an account (only company’s clients have one) in order to see them. But password reset wizard is technically very similar to the internal ones, so, if you are able to efficiently block it, I may adjust the rule(s) for them too.

**Reply by D4niloMR** [reply]
These are for log in the console to see if it has info related to the popups.

You can try one by one, maybe it catchs the popup.
```
tools.otenet.gr##+js(aeld, , , log, 3)
tools.otenet.gr##+js(nostif, /^/)
tools.otenet.gr##+js(nosiif, /^/)
tools.otenet.gr##+js(aeld, /click|mousedown/, , log, 1)
tools.otenet.gr##+js(nostif, popup)
tools.otenet.gr##+js(nosiif, popup)
```

set `filterAuthorMode` to `true` in advanced settings and search for specific keywords in the console.

also with https://github.com/uBlockOrigin/uAssets/discussions/16400#discussioncomment-8184710

**Comment by DimitrisKalaboukas** [comment, marked as answer]
The site changed design and the issue has gone. So, the rules:

tools.otenet.gr##.ui-dialog:has(#oldPasswordPromptMsg)
tools.otenet.gr##.ui-dialog:has(#pr_popupwizard)
tools.otenet.gr##.ui-widget-overlay

are now sufficient.
